# Objective of the Analysis
#### This analysis investigates the Google Play Store dataset from multiple perspectives to understand app behavior, user engagement, and market trends.

In [2]:
import pandas as pd
import numpy as np



In [3]:
df = pd.read_csv("googleplaystore 2.csv")

In [4]:
df.head()

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
0,Photo Editor & Candy Camera & Grid & ScrapBook,ART_AND_DESIGN,4.1,159,19M,"10,000+",Free,0,Everyone,Art & Design,"January 7, 2018",1.0.0,4.0.3 and up
1,Coloring book moana,ART_AND_DESIGN,3.9,967,14M,"500,000+",Free,0,Everyone,Art & Design;Pretend Play,"January 15, 2018",2.0.0,4.0.3 and up
2,"U Launcher Lite – FREE Live Cool Themes, Hide ...",ART_AND_DESIGN,4.7,87510,8.7M,"5,000,000+",Free,0,Everyone,Art & Design,"August 1, 2018",1.2.4,4.0.3 and up
3,Sketch - Draw & Paint,ART_AND_DESIGN,4.5,215644,25M,"50,000,000+",Free,0,Teen,Art & Design,"June 8, 2018",Varies with device,4.2 and up
4,Pixel Draw - Number Art Coloring Book,ART_AND_DESIGN,4.3,967,2.8M,"100,000+",Free,0,Everyone,Art & Design;Creativity,"June 20, 2018",1.1,4.4 and up


In [5]:
df.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='str')

In [6]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  str    
 1   Category        10841 non-null  str    
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  str    
 4   Size            10841 non-null  str    
 5   Installs        10841 non-null  str    
 6   Type            10840 non-null  str    
 7   Price           10841 non-null  str    
 8   Content Rating  10840 non-null  str    
 9   Genres          10841 non-null  str    
 10  Last Updated    10841 non-null  str    
 11  Current Ver     10833 non-null  str    
 12  Android Ver     10838 non-null  str    
dtypes: float64(1), str(12)
memory usage: 1.1 MB


In [7]:
df.describe()

,Rating
count,9367.000000
mean,4.193338
std,0.537431
min,1.000000
25%,4.000000
50%,4.300000
75%,4.500000
max,19.000000


In [8]:
df.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

In [ ]:
df.duplicated().sum() #represents teh number of Duplicated cells

np.int64(483)

In [59]:
df.drop_duplicates(inplace=True) # Remove duplicate rows to avoid bias in analysis

In [11]:
df.duplicated().sum()

np.int64(0)

In [12]:
df["Installs"].head()

0        10,000+
1       500,000+
2     5,000,000+
3    50,000,000+
4       100,000+
Name: Installs, dtype: str

In [13]:
df["Installs"] = df["Installs"].str.replace(",", "")
df["Installs"] = df["Installs"].str.replace("+", "")

In [14]:
df.head()

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
0,Photo Editor & Candy Camera & Grid & ScrapBook,ART_AND_DESIGN,4.1,159,19M,10000,Free,0,Everyone,Art & Design,"January 7, 2018",1.0.0,4.0.3 and up
1,Coloring book moana,ART_AND_DESIGN,3.9,967,14M,500000,Free,0,Everyone,Art & Design;Pretend Play,"January 15, 2018",2.0.0,4.0.3 and up
2,"U Launcher Lite – FREE Live Cool Themes, Hide ...",ART_AND_DESIGN,4.7,87510,8.7M,5000000,Free,0,Everyone,Art & Design,"August 1, 2018",1.2.4,4.0.3 and up
3,Sketch - Draw & Paint,ART_AND_DESIGN,4.5,215644,25M,50000000,Free,0,Teen,Art & Design,"June 8, 2018",Varies with device,4.2 and up
4,Pixel Draw - Number Art Coloring Book,ART_AND_DESIGN,4.3,967,2.8M,100000,Free,0,Everyone,Art & Design;Creativity,"June 20, 2018",1.1,4.4 and up


In [15]:
df.tail()

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver
10836,Sya9a Maroc - FR,FAMILY,4.5,38,53M,5000,Free,0,Everyone,Education,"July 25, 2017",1.48,4.1 and up
10837,Fr. Mike Schmitz Audio Teachings,FAMILY,5.0,4,3.6M,100,Free,0,Everyone,Education,"July 6, 2018",1,4.1 and up
10838,Parkinson Exercices FR,MEDICAL,NaN,3,9.5M,1000,Free,0,Everyone,Medical,"January 20, 2017",1,2.2 and up
10839,The SCP Foundation DB fr nn5n,BOOKS_AND_REFERENCE,4.5,114,Varies with device,1000,Free,0,Mature 17+,Books & Reference,"January 19, 2015",Varies with device,Varies with device
10840,iHoroscope - 2018 Daily Horoscope & Astrology,LIFESTYLE,4.5,398307,19M,10000000,Free,0,Everyone,Lifestyle,"July 25, 2018",Varies with device,Varies with device


In [16]:
df['Installs'] = pd.to_numeric(df['Installs'], errors="coerce")

In [17]:
df["Price"] = df["Price"].str.replace("$", "")
df["Price"] = pd.to_numeric(df["Price"], errors="coerce")

In [18]:
df["Reviews"] = df["Reviews"].str.replace("M", "")
df["Reviews"] = pd.to_numeric(df["Reviews"], errors="coerce")

In [19]:
df.info()

<class 'pandas.DataFrame'>
Index: 10358 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10358 non-null  str    
 1   Category        10358 non-null  str    
 2   Rating          8893 non-null   float64
 3   Reviews         10358 non-null  float64
 4   Size            10358 non-null  str    
 5   Installs        10357 non-null  float64
 6   Type            10357 non-null  str    
 7   Price           10357 non-null  float64
 8   Content Rating  10357 non-null  str    
 9   Genres          10358 non-null  str    
 10  Last Updated    10358 non-null  str    
 11  Current Ver     10350 non-null  str    
 12  Android Ver     10355 non-null  str    
dtypes: float64(4), str(9)
memory usage: 1.1 MB


In [20]:
df.isnull().sum()

App                  0
Category             0
Rating            1465
Reviews              0
Size                 0
Installs             1
Type                 1
Price                1
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

In [21]:
df.isna().sum()

App                  0
Category             0
Rating            1465
Reviews              0
Size                 0
Installs             1
Type                 1
Price                1
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

In [22]:
df.info()

<class 'pandas.DataFrame'>
Index: 10358 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10358 non-null  str    
 1   Category        10358 non-null  str    
 2   Rating          8893 non-null   float64
 3   Reviews         10358 non-null  float64
 4   Size            10358 non-null  str    
 5   Installs        10357 non-null  float64
 6   Type            10357 non-null  str    
 7   Price           10357 non-null  float64
 8   Content Rating  10357 non-null  str    
 9   Genres          10358 non-null  str    
 10  Last Updated    10358 non-null  str    
 11  Current Ver     10350 non-null  str    
 12  Android Ver     10355 non-null  str    
dtypes: float64(4), str(9)
memory usage: 1.1 MB


In [23]:
df['Last Updated'] = pd.to_datetime

In [24]:
df['Android_Ver_Num'] = df['Android Ver'].str.extract('(\d+\.\d+)')
df['Android_Ver_Num'] = pd.to_numeric(df['Android_Ver_Num'], errors='coerce')

In [25]:
df.head()

,App,Category,Rating,Reviews,Size,Installs,Type,Price,Content Rating,Genres,Last Updated,Current Ver,Android Ver,Android_Ver_Num
0,Photo Editor & Candy Camera & Grid & ScrapBook,ART_AND_DESIGN,4.1,159.0,19M,10000.0,Free,0.0,Everyone,Art & Design,<function to_datetime at 0x118062f20>,1.0.0,4.0.3 and up,4.0
1,Coloring book moana,ART_AND_DESIGN,3.9,967.0,14M,500000.0,Free,0.0,Everyone,Art & Design;Pretend Play,<function to_datetime at 0x118062f20>,2.0.0,4.0.3 and up,4.0
2,"U Launcher Lite – FREE Live Cool Themes, Hide ...",ART_AND_DESIGN,4.7,87510.0,8.7M,5000000.0,Free,0.0,Everyone,Art & Design,<function to_datetime at 0x118062f20>,1.2.4,4.0.3 and up,4.0
3,Sketch - Draw & Paint,ART_AND_DESIGN,4.5,215644.0,25M,50000000.0,Free,0.0,Teen,Art & Design,<function to_datetime at 0x118062f20>,Varies with device,4.2 and up,4.2
4,Pixel Draw - Number Art Coloring Book,ART_AND_DESIGN,4.3,967.0,2.8M,100000.0,Free,0.0,Everyone,Art & Design;Creativity,<function to_datetime at 0x118062f20>,1.1,4.4 and up,4.4


In [26]:
df['Rating'].describe()

count    8893.000000
mean        4.189542
std         0.545452
min         1.000000
25%         4.000000
50%         4.300000
75%         4.500000
max        19.000000
Name: Rating, dtype: float64

In [27]:
df['Rating'] = df['Rating'].fillna(df['Rating'].median())

In [28]:
df['Type'].fillna('Free', inplace=True)

/var/folders/47/t6ys7c_56jsgwg3stl54nxfr0000gn/T/ipykernel_796/1900772705.py:1: ChainedAssignmentError: A value is being set on a copy of a DataFrame or Series through chained assignment using an inplace method.
Such inplace method never works to update the original DataFrame or Series, because the intermediate object on which we are setting values always behaves as a copy (due to Copy-on-Write).

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' instead, to perform the operation inplace on the original object, or try to avoid an inplace operation using 'df[col] = df[col].method(value)'.

See the documentation for a more detailed explanation: https://pandas.pydata.org/pandas-docs/stable/user_guide/copy_on_write.html
  df['Type'].fillna('Free', inplace=True)


0        Free
1        Free
2        Free
3        Free
4        Free
         ... 
10836    Free
10837    Free
10838    Free
10839    Free
10840    Free
Name: Type, Length: 10358, dtype: str

In [29]:
def clean_size(x):
    if isinstance(x, str):
        x = x.strip()
        
        if 'M' in x:
            return float(x.replace('M',''))
        elif 'k' in x:
            return float(x.replace('k','')) / 1024
        else:
            return np.nan
    
    elif isinstance(x, (int, float)):
        return x  # keep numeric values as they are
    
    return np.nan

df['Size'] = df['Size'].replace('Varies with device', np.nan)
df['Size'] = df['Size'].apply(clean_size)

In [30]:
df.info()
df.isnull().sum()

<class 'pandas.DataFrame'>
Index: 10358 entries, 0 to 10840
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   App              10358 non-null  str    
 1   Category         10358 non-null  str    
 2   Rating           10358 non-null  float64
 3   Reviews          10358 non-null  float64
 4   Size             8831 non-null   float64
 5   Installs         10357 non-null  float64
 6   Type             10357 non-null  str    
 7   Price            10357 non-null  float64
 8   Content Rating   10357 non-null  str    
 9   Genres           10358 non-null  str    
 10  Last Updated     10358 non-null  object 
 11  Current Ver      10350 non-null  str    
 12  Android Ver      10355 non-null  str    
 13  Android_Ver_Num  9134 non-null   float64
dtypes: float64(6), object(1), str(7)
memory usage: 1.2+ MB


App                   0
Category              0
Rating                0
Reviews               0
Size               1527
Installs              1
Type                  1
Price                 1
Content Rating        1
Genres                0
Last Updated          0
Current Ver           8
Android Ver           3
Android_Ver_Num    1224
dtype: int64

In [31]:
df['Category'].value_counts().head(10)

Category
FAMILY             1943
GAME               1121
TOOLS               843
BUSINESS            427
MEDICAL             408
PRODUCTIVITY        407
PERSONALIZATION     388
LIFESTYLE           373
COMMUNICATION       366
FINANCE             360
Name: count, dtype: int64

## The General Aim is to identify:
### What factors are associated with app popularity on the Google Play Store (installs, reviews, and ratings)?

## Free Vs Paid Comparison
Do paid apps have higher ratings than free apps?

In [32]:
df.groupby('Type')['Rating'].mean()

Type
0       19.000000
Free     4.198509
Paid     4.269150
Name: Rating, dtype: float64

In [33]:

import plotly.express as px


In [34]:
df.info()

<class 'pandas.DataFrame'>
Index: 10358 entries, 0 to 10840
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   App              10358 non-null  str    
 1   Category         10358 non-null  str    
 2   Rating           10358 non-null  float64
 3   Reviews          10358 non-null  float64
 4   Size             8831 non-null   float64
 5   Installs         10357 non-null  float64
 6   Type             10357 non-null  str    
 7   Price            10357 non-null  float64
 8   Content Rating   10357 non-null  str    
 9   Genres           10358 non-null  str    
 10  Last Updated     10358 non-null  object 
 11  Current Ver      10350 non-null  str    
 12  Android Ver      10355 non-null  str    
 13  Android_Ver_Num  9134 non-null   float64
dtypes: float64(6), object(1), str(7)
memory usage: 1.2+ MB


In [36]:
df["Size"].fillna(df["Size"].median(), inplace=True)

/var/folders/47/t6ys7c_56jsgwg3stl54nxfr0000gn/T/ipykernel_796/3314193432.py:1: ChainedAssignmentError: A value is being set on a copy of a DataFrame or Series through chained assignment using an inplace method.
Such inplace method never works to update the original DataFrame or Series, because the intermediate object on which we are setting values always behaves as a copy (due to Copy-on-Write).

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' instead, to perform the operation inplace on the original object, or try to avoid an inplace operation using 'df[col] = df[col].method(value)'.

See the documentation for a more detailed explanation: https://pandas.pydata.org/pandas-docs/stable/user_guide/copy_on_write.html
  df["Size"].fillna(df["Size"].median(), inplace=True)


0        19.0
1        14.0
2         8.7
3        25.0
4         2.8
         ... 
10836    53.0
10837     3.6
10838     9.5
10839    13.0
10840    19.0
Name: Size, Length: 10358, dtype: float64

# Q1 Which Categories have the most Apps? (Univariate)

In [38]:
fig = px.histogram(
    df,
    y="Category",
    title="Distribution of App Categories"
)

fig.show()

#### Extra figure

In [51]:
fig = px.treemap(
    df,
    path=["Category"],
    title="Category"
)

fig.show()

Obviosuly The family categories have the highest number of apps with significant difference than other apps, Therfore the family catergory dominates then Game comes after

# Q2 What is the distribution of app ratings on the Google Play Store? (Univariate)

In [35]:

fig = px.histogram(df, x="Rating")
fig.show()

4.3 has the highest rating, the value itself is not a strong indicator but the highest count of ratings are around 4 ndicating generally positive user feedback.

# Q3 Does The price affect the app Rating? (Bi-Variate)

In [78]:
fig = px.scatter(
    df,
    x="Price",
    y="Rating",
    title="Price vs Rating",
    hover_data=["App"]
)

fig.show()

I see that the ratings are affected by the price but not highly affected 
I see that the lowest the price the more users give ratings but there are few people that gave high ratings to high price games

# Q4 Are most common apps Free or Paid? (UniVariate)

In [41]:
fig = px.pie(
    df,
    names="Type",
    title="Free vs Paid Apps"
)

fig.show()

It seems that 92.6% of apps are free , so the probability of free apps are much higher that the paid one

### Extra Figure

In [58]:
fig = px.histogram(
    df,
    x ="Type",
    title="Distribution of App Categories"
)

fig.show()

# Q5 Is there a Direct corerelation between Rating and Installs? (BiVariate)

In [79]:
fig = px.scatter(df, x="Rating", y="Installs", title="Rating vs Installs")
fig.show()

There is no strong direct correlation, But the Highest number of installs has above 3 ratings

# Q6 How does Android version affect app ratings and user engagement on the Google Play Store? (BiVariate)

In [62]:
fig = px.box(
    df,
    x="Android_Ver_Num",
    y="Installs",
    title="App Installs Across Android Versions"
)

fig.show()

There is no significant correlation between the android version and the number of installs with a weak bias to the higher versions

# Q7 Are people triggered by the size of the app? so do the number of installs decrease as the app size increase?

In [72]:
px.scatter(df, x='Size', y='Installs')

App size does not significantly affect the number of installs, since apps of different sizes can achieve similar install levels.

## Q8 Is There a Relation betweeing the Ratings and the size of the Apps? (BiVariate)

In [77]:
px.scatter(df, x='Rating', y='Size')

There Is no strong correlation between Rating and size

## Q9 Which Categories have the largest number of Installs? (BiVariate)

In [88]:


fig = px.histogram(
    df,
    x='Category',
    y='Installs',
    title='Category vs Installs'
   
)

fig.show()

Family and Communication has the largest number of Installs


## Q10 What are the average of the app sizes?

In [96]:

fig = px.histogram(
    df,
    x='Size',
    title='Distribution of App Size',
    nbins=30 #30 bars
)

fig.show()

Most apps sizes are within range 2.5-7.4 MB

## 6. Summary of Findings

| # | Question | Key Finding |
|---|----------|-------------|
| 1 | Category distribution | FAMILY has the most apps; GAME & TOOLS follow |
| 2 | Rating distribution | Left-skewed; most apps rated 4.0–4.5 |
| 3 | Free vs Paid | ~93% of apps are Free |
| 4 | App size |  Most apps fall within size 2.5-7.4|
| 5 | Price ↔ Rating | Very weak positive relationship |
| 6 | Rating ↔ Installs | No strong correlation; Reviews & Installs are |
| 7 | Android version ↔ Installs | Broader compatibility (newer versions) → more installs |
| 8 | Size ↔ Installs | No significant relationship |
| 9 | Category ↔ Installs | Communication and Family lead total installs |


